### PR TITLE
[PMON] [Mellanox] fix syseepromd issue on simx

### PR DESCRIPTION
**Pull request by tomer-israel**
[PMON] syseepromd fix for  mellanox simulator platform - avoid initializing sfp/thermal/components/fan/psu/leds on simx and create vpd_info file on hw_management when we use mellanox simulator platform

#### Why I did it
this is a fix for issue in mellanox simulator platforms. the syseepromd failed on the pmon docker. also "decode-syseeprom" failed also

#### How I did it
- before initializing thermal/components/fan/psu/leds --> check if we are running on simx
- creating the vpd_info on the hw_management folder.

#### How to verify it
- check if syseepromd process was loaded properly on the pmon docker.
- decode-syseeprom is working well without errors/warnings

#### Which release branch to backport (provide reason below if selected)
- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012

#### Description for the changelog
fix issues on syseeprom on mellanox simulator platforms. problem was discovered on 202012 release.

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@nazariig could you please help to review?

File: platform/mellanox/mlnx-platform-api/sonic_platform/eeprom.py [hunk 1]
```diff
@@ -25,6 +26,18 @@
 #
 EEPROM_SYMLINK = "/var/run/hw-management/eeprom/vpd_info"
 
+platform_name = get_platform()
+if 'simx' in platform_name:
+    if is_host():
```
Comment (nazariig) [inline platform/mellanox/mlnx-platform-api/sonic_platform/eeprom.py:31]: @tomer-israel why not to use `get_path_to_platform_dir`?

File: platform/mellanox/mlnx-platform-api/sonic_platform/platform.py [hunk 2]
```diff
@@ -15,11 +16,14 @@ class Platform(PlatformBase):
...
-        self._chassis.initialize_psu()
         self._chassis.initialize_eeprom()
-        if utils.is_host():
-            self._chassis.initialize_components()
-            self._chassis.initizalize_system_led()
-        else:
-            self._chassis.initialize_fan()
-            self._chassis.initialize_thermals()
+        platform_name = get_platform()
+        simx_platform = bool(platform_name and "simx" in platform_name)
```
Comment (nazariig) [inline platform/mellanox/mlnx-platform-api/sonic_platform/platform.py:21]: @tomer-israel why do we need `bool()`?

File: platform/mellanox/mlnx-platform-api/sonic_platform/eeprom.py (hunk 1 above)
Comment (tomer-israel) [reply platform/mellanox/mlnx-platform-api/sonic_platform/eeprom.py]: i used get_path_to_platform_dir() instead

File: platform/mellanox/mlnx-platform-api/sonic_platform/platform.py (hunk 2 above)
Comment (tomer-israel) [reply platform/mellanox/mlnx-platform-api/sonic_platform/platform.py]: I changed it, now it checks that "simx" not exist on platform_name

File: platform/mellanox/mlnx-platform-api/sonic_platform/eeprom.py [hunk 3]
```diff
@@ -25,6 +26,15 @@
...
 EEPROM_SYMLINK = "/var/run/hw-management/eeprom/vpd_info"
 
+platform_name = get_platform()
+if 'simx' in platform_name:
+    platform_path = get_path_to_platform_dir()
+
+    if not os.path.exists(os.path.dirname(EEPROM_SYMLINK)):
+        os.makedirs(os.path.dirname(EEPROM_SYMLINK))
+    if not os.path.exists(EEPROM_SYMLINK):
+        subprocess.check_call(['/usr/bin/xxd', '-r', '-p', 'syseeprom.hex', EEPROM_SYMLINK], cwd=platform_path)
```
Comment (nazariig) [inline platform/mellanox/mlnx-platform-api/sonic_platform/eeprom.py:36]: @tomer-israel why do we need to always check both?

Example:
```cpp
if not os.path.exists(EEPROM_SYMLINK):
    if not os.path.exists(os.path.dirname(EEPROM_SYMLINK)):
        os.makedirs(os.path.dirname(EEPROM_SYMLINK))

    subprocess.check_call(['/usr/bin/xxd', '-r', '-p', 'syseeprom.hex', EEPROM_SYMLINK], cwd=platform_path)
```

Comment (tomer-israel) [reply platform/mellanox/mlnx-platform-api/sonic_platform/eeprom.py:37]: changed it as you suggested

**Review by nazariig — APPROVED** [review]

**Review by lguohan — APPROVED** [review]

**Comment by qiluo-msft** [comment]
This PR could not be cleanly cherry-pick to 202012. Please submit another PR.

**Pull request merged**